contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -170,6 +176,25 @@ func (t *T) Has(f FlagType) (exists bool) {
 	return
 }
 
+func (t *T) Roles() []string {
+	if len(t.Pod.ObjectMeta.Labels) == 0 {
+		return defaultRoles
+	}
+
+	roles := strings.Split(t.Pod.ObjectMeta.Labels[annotation.RolesKey], ",")
+	for i, r := range roles {
+		roles[i] = strings.TrimSpace(r)
+	}
+
+	roles = filterRoles(roles, not(emptyRole), not(duplicateRole()))
```

Comment:
Hmm, not following, the map is built on the fly during filter evaluation, there is no separate build step. duplicateRole returns a func closing over an empty map.